providers/os/connection/device/device_connection.go

```diff
@@ -83,12 +96,29 @@ func NewDeviceConnection(connId uint32, conf *inventory.Config, asset *inventory
 	// we iterate over all the blocks and try to run OS detection on each one of them
 	// we only return one asset, if we find the right block (e.g. the one with the root FS)
 	for _, block := range blocks {
-		fsConn, scanDir, err := tryDetectAsset(connId, block, manager, conf, asset)
-		if scanDir != "" {
-			res.MountedDirs = append(res.MountedDirs, scanDir)
-			res.partitions[scanDir] = block
+		log.Debug().
+			Str("name", block.Name).
+			Str("type", block.FsType).
+			Str("mountpoint", block.MountPoint).
+			Msg("trying partition for asset detection")
+
+		if block.MountPoint == "" {
+			scanDir, err := manager.Mount(block)
+			if err != nil {
+				log.Error().Err(err).Msg("unable to complete mount step")
+				continue
+			}
+			block.MountPoint = scanDir
+		}
+		res.MountedDirs = append(res.MountedDirs, block.MountPoint)
```

Comment:
should we only append this if the block was mounted by us? is that then used for unmounting?

Reply:
This is used by other providers that use device connection indirectly, i.e. without fs connection.